packages/devtools_app/lib/src/debugger/file_search.dart

```diff
@@ -0,0 +1,146 @@
+  final Map<String, ScriptRef> _scriptsCache = {};
+
+  final fileSearchFieldKey = GlobalKey(debugLabel: 'fileSearchFieldKey');
+
+  @override
+  void initState() {
+    super.initState();
+
+    _autoCompleteController = AutoCompleteController();
+    _autoCompleteController.currentDefaultIndex = 0;
+
+    addAutoDisposeListener(
+        _autoCompleteController.selectTheSearchNotifier, _handleSearch);
+    addAutoDisposeListener(
+        _autoCompleteController.searchNotifier, _handleSearch);
+
+    _autoCompleteController.selectTheSearch = true;
+  }
+
+  void _handleSearch() {
+    final query = _autoCompleteController.search;
+    final matches = findMatches(query, widget.controller.sortedScripts.value);
+    matches.forEach(_addScriptRefToCache);
+    _autoCompleteController.searchAutoComplete.value =
+        matches.map((scriptRef) => scriptRef.uri).toList();
+    _handleAutoCompleteOverlay();
+  }
+
+  void _handleAutoCompleteOverlay() {
+    _autoCompleteController.handleAutoCompleteOverlay(
+      context: context,
+      searchFieldKey: fileSearchFieldKey,
+      onTap: _onSelection,
+    );
+  }
+
+  void _addScriptRefToCache(ScriptRef scriptRef) {
+    if (!_scriptsCache.containsKey(scriptRef.uri)) {
+      _scriptsCache[scriptRef.uri] = scriptRef;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return Container(
+      decoration: BoxDecoration(
+        border: Border(
+          top: BorderSide(color: theme.focusColor),
+        ),
+      ),
+      padding: const EdgeInsets.all(8.0),
+      child: Row(
+        children: [
+          const SizedBox(width: 8.0),
+          Expanded(
+            child: buildAutoCompleteSearchField(
+              controller: _autoCompleteController,
+              searchFieldKey: fileSearchFieldKey,
+              searchFieldEnabled: true,
+              shouldRequestFocus: true,
+              closeOverlayOnEscape: false,
+              onSelection: _onSelection,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void _onSelection(String scriptUri) {
+    final scriptRef = _scriptsCache[scriptUri];
+    widget.controller.showScriptLocation(ScriptLocation(scriptRef));
+    _scriptsCache.clear();
+    widget.handleClose();
+  }
+
+  @override
+  void dispose() {
+    _autoCompleteController.dispose();
+    super.dispose();
+  }
+}
+
+List<ScriptRef> findMatches(
+  String query,
+  List<ScriptRef> scriptRefs,
+) {
+  if (query.isEmpty) {
+    takeTopMatches(scriptRefs);
+  }
+
+  final exactMatches = scriptRefs
+      .where((scriptRef) => scriptRef.uri.caseInsensitiveContains(query))
+      .toList();
+
+  if (exactMatches.length >= numOfMatchesToShow) {
+    return takeTopMatches(exactMatches);
+  }
+
+  final fuzzyMatches = scriptRefs
+      .where((scriptRef) => scriptRef.uri.caseInsensitiveFuzzyMatch(query))
+      .toList();
+
+  return takeTopMatches([...exactMatches, ...fuzzyMatches, ...scriptRefs]);
```

Comment:
should we still be including scriptRefs here? If none match the query, I think we should have empty results. This matches the open file behavior in IntelliJ.

Reply:
Good point, changed so that we return don't return any files if there is no match